Failing tests log

<details>

```
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Using setuptoolsShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing setuptools-check-hook
Using setuptoolsCheckPhase
Sourcing pytest-check-hook
Using pytestCheckPhase
Removing setuptoolsCheckPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/ywhyp0rfnljl8547f00wxi1qrnq1mrk2-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/tests/test_utils.py
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Executing setuptoolsBuildPhase
running bdist_wheel
running build
running build_py
creating build
creating build/lib
creating build/lib/aocd
copying aocd/cookies.py -> build/lib/aocd
copying aocd/post.py -> build/lib/aocd
copying aocd/transforms.py -> build/lib/aocd
copying aocd/runner.py -> build/lib/aocd
copying aocd/version.py -> build/lib/aocd
copying aocd/models.py -> build/lib/aocd
copying aocd/__init__.py -> build/lib/aocd
copying aocd/cli.py -> build/lib/aocd
copying aocd/utils.py -> build/lib/aocd
copying aocd/exceptions.py -> build/lib/aocd
copying aocd/get.py -> build/lib/aocd
installing to build/bdist.macosx-10.6-x86_64/wheel
running install
running install_lib
creating build/bdist.macosx-10.6-x86_64
creating build/bdist.macosx-10.6-x86_64/wheel
creating build/bdist.macosx-10.6-x86_64/wheel/aocd
copying build/lib/aocd/cookies.py -> build/bdist.macosx-10.6-x86_64/wheel/aocd
copying build/lib/aocd/post.py -> build/bdist.macosx-10.6-x86_64/wheel/aocd
copying build/lib/aocd/transforms.py -> build/bdist.macosx-10.6-x86_64/wheel/aocd
copying build/lib/aocd/runner.py -> build/bdist.macosx-10.6-x86_64/wheel/aocd
copying build/lib/aocd/version.py -> build/bdist.macosx-10.6-x86_64/wheel/aocd
copying build/lib/aocd/models.py -> build/bdist.macosx-10.6-x86_64/wheel/aocd
copying build/lib/aocd/__init__.py -> build/bdist.macosx-10.6-x86_64/wheel/aocd
copying build/lib/aocd/cli.py -> build/bdist.macosx-10.6-x86_64/wheel/aocd
copying build/lib/aocd/utils.py -> build/bdist.macosx-10.6-x86_64/wheel/aocd
copying build/lib/aocd/exceptions.py -> build/bdist.macosx-10.6-x86_64/wheel/aocd
copying build/lib/aocd/get.py -> build/bdist.macosx-10.6-x86_64/wheel/aocd
running install_egg_info
running egg_info
creating advent_of_code_data.egg-info
writing advent_of_code_data.egg-info/PKG-INFO
writing dependency_links to advent_of_code_data.egg-info/dependency_links.txt
writing entry points to advent_of_code_data.egg-info/entry_points.txt
writing requirements to advent_of_code_data.egg-info/requires.txt
writing top-level names to advent_of_code_data.egg-info/top_level.txt
writing manifest file 'advent_of_code_data.egg-info/SOURCES.txt'
reading manifest file 'advent_of_code_data.egg-info/SOURCES.txt'
adding license file 'LICENSE'
writing manifest file 'advent_of_code_data.egg-info/SOURCES.txt'
Copying advent_of_code_data.egg-info to build/bdist.macosx-10.6-x86_64/wheel/advent_of_code_data-1.1.1-py3.9.egg-info
running install_scripts
adding license file "LICENSE" (matched pattern "LICEN[CS]E*")
creating build/bdist.macosx-10.6-x86_64/wheel/advent_of_code_data-1.1.1.dist-info/WHEEL
creating 'dist/advent_of_code_data-1.1.1-py2.py3-none-any.whl' and adding 'build/bdist.macosx-10.6-x86_64/wheel' to it
adding 'aocd/__init__.py'
adding 'aocd/cli.py'
adding 'aocd/cookies.py'
adding 'aocd/exceptions.py'
adding 'aocd/get.py'
adding 'aocd/models.py'
adding 'aocd/post.py'
adding 'aocd/runner.py'
adding 'aocd/transforms.py'
adding 'aocd/utils.py'
adding 'aocd/version.py'
adding 'advent_of_code_data-1.1.1.dist-info/LICENSE'
adding 'advent_of_code_data-1.1.1.dist-info/METADATA'
adding 'advent_of_code_data-1.1.1.dist-info/WHEEL'
adding 'advent_of_code_data-1.1.1.dist-info/entry_points.txt'
adding 'advent_of_code_data-1.1.1.dist-info/top_level.txt'
adding 'advent_of_code_data-1.1.1.dist-info/RECORD'
removing build/bdist.macosx-10.6-x86_64/wheel
Finished executing setuptoolsBuildPhase
@nix { "action": "setPhase", "phase": "installPhase" }
installing
Executing pipInstallPhase
/private/tmp/nix-build-python3.9-aocd-1.1.1.drv-0/source/dist /private/tmp/nix-build-python3.9-aocd-1.1.1.drv-0/source
Processing ./advent_of_code_data-1.1.1-py2.py3-none-any.whl
Requirement already satisfied: pebble in /nix/store/43rjwqdx6zk19qirab1213zri48vxgk1-python3.9-pebble-4.6.0/lib/python3.9/site-packages (from advent-of-code-data==1.1.1) (4.6.0)
Requirement already satisfied: tzlocal in /nix/store/k4qdapvjx7cncw3kdyvvp8gngys5ikv8-python3.9-tzlocal-4.1/lib/python3.9/site-packages (from advent-of-code-data==1.1.1) (4.1)
Requirement already satisfied: termcolor in /nix/store/mwjp2yz16pjx1dn0wx14i45ablcxnvy5-python3.9-termcolor-1.1.0/lib/python3.9/site-packages (from advent-of-code-data==1.1.1) (1.1.0)
Requirement already satisfied: requests in /nix/store/0j3wzs714cx8msw0l6pq0i4yds9pgmiq-python3.9-requests-2.26.0/lib/python3.9/site-packages (from advent-of-code-data==1.1.1) (2.26.0)
Requirement already satisfied: beautifulsoup4 in /nix/store/xyp7cdrwhmpv7jf2prnfcw809wp7m4qj-python3.9-beautifulsoup4-4.10.0/lib/python3.9/site-packages (from advent-of-code-data==1.1.1) (4.10.0)
Requirement already satisfied: python-dateutil in /nix/store/s5l636d48pc7n1wh384i7r0gg2nblsvl-python3.9-python-dateutil-2.8.2/lib/python3.9/site-packages (from advent-of-code-data==1.1.1) (2.8.2)
Requirement already satisfied: soupsieve>1.2 in /nix/store/9innc1jlbc8hsn5vh98dj37rwxgjsw5c-python3.9-soupsieve-2.2.1/lib/python3.9/site-packages (from beautifulsoup4->advent-of-code-data==1.1.1) (2.2.1)
Requirement already satisfied: six>=1.5 in /nix/store/n526dasm15zlba2q9qxhjmflyjy8kfsh-python3.9-six-1.16.0/lib/python3.9/site-packages (from python-dateutil->advent-of-code-data==1.1.1) (1.16.0)
Requirement already satisfied: certifi>=2017.4.17 in /nix/store/6jqa1ia6b6qvjvncjfi7mxas76ghrywv-python3.9-certifi-2021.10.08/lib/python3.9/site-packages (from requests->advent-of-code-data==1.1.1) (2021.10.8)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /nix/store/bgv7nfxq66mfm8b9fj8k6mxwpw06q3lr-python3.9-urllib3-1.26.7/lib/python3.9/site-packages (from requests->advent-of-code-data==1.1.1) (1.26.7)
Requirement already satisfied: idna<4,>=2.5 in /nix/store/hq65wwp5pkgqh8jn28rv3cqgqq98l3gv-python3.9-idna-3.2/lib/python3.9/site-packages (from requests->advent-of-code-data==1.1.1) (3.2)
Requirement already satisfied: charset-normalizer~=2.0.0 in /nix/store/2npr123zi0g5b2k98d7s549pmx06dhb1-python3.9-charset-normalizer-2.0.5/lib/python3.9/site-packages (from requests->advent-of-code-data==1.1.1) (2.0.5)
Requirement already satisfied: pytz-deprecation-shim in /nix/store/cgvq0rrbd4qnq6wf23c3g96221bsxz4y-python3.9-pytz-deprecation-shim-0.1.0.post0/lib/python3.9/site-packages (from tzlocal->advent-of-code-data==1.1.1) (0.1.0.post0)
Requirement already satisfied: tzdata in /nix/store/hk8d4zfbbsnghj33x9k8dfl5sll1wxqi-python3.9-tzdata-2021.1/lib/python3.9/site-packages (from pytz-deprecation-shim->tzlocal->advent-of-code-data==1.1.1) (2021.1)
Installing collected packages: advent-of-code-data
Successfully installed advent-of-code-data-1.1.1
/private/tmp/nix-build-python3.9-aocd-1.1.1.drv-0/source
Finished executing pipInstallPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
post-installation fixup
strip is /nix/store/854jyvxrvpdpbfn2zaba1v2qgqkxipyh-cctools-binutils-darwin-949.0.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/3qjxnkn6q5vgq0yjpsbkl7lzqj8504ml-python3.9-aocd-1.1.1/lib  /nix/store/3qjxnkn6q5vgq0yjpsbkl7lzqj8504ml-python3.9-aocd-1.1.1/bin 
patching script interpreter paths in /nix/store/3qjxnkn6q5vgq0yjpsbkl7lzqj8504ml-python3.9-aocd-1.1.1
Rewriting #!/nix/store/w8lqcsrwlwrvmchj46ph9880w9h4m50a-python3-3.9.6/bin/python3.9 to #!/nix/store/w8lqcsrwlwrvmchj46ph9880w9h4m50a-python3-3.9.6
wrapping `/nix/store/3qjxnkn6q5vgq0yjpsbkl7lzqj8504ml-python3.9-aocd-1.1.1/bin/aoc'...
Rewriting #!/nix/store/w8lqcsrwlwrvmchj46ph9880w9h4m50a-python3-3.9.6/bin/python3.9 to #!/nix/store/w8lqcsrwlwrvmchj46ph9880w9h4m50a-python3-3.9.6
wrapping `/nix/store/3qjxnkn6q5vgq0yjpsbkl7lzqj8504ml-python3.9-aocd-1.1.1/bin/aocd'...
Rewriting #!/nix/store/w8lqcsrwlwrvmchj46ph9880w9h4m50a-python3-3.9.6/bin/python3.9 to #!/nix/store/w8lqcsrwlwrvmchj46ph9880w9h4m50a-python3-3.9.6
wrapping `/nix/store/3qjxnkn6q5vgq0yjpsbkl7lzqj8504ml-python3.9-aocd-1.1.1/bin/aocd-token'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
@nix { "action": "setPhase", "phase": "installCheckPhase" }
running install tests
no Makefile or custom installCheckPhase, doing nothing
@nix { "action": "setPhase", "phase": "pythonCatchConflictsPhase" }
pythonCatchConflictsPhase
@nix { "action": "setPhase", "phase": "pythonRemoveBinBytecodePhase" }
pythonRemoveBinBytecodePhase
@nix { "action": "setPhase", "phase": "pythonImportsCheckPhase" }
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: aocd
@nix { "action": "setPhase", "phase": "pytestCheckPhase" }
pytestCheckPhase
Executing pytestCheckPhase
[1m============================= test session starts ==============================[0m
platform darwin -- Python 3.9.6, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
rootdir: /private/tmp/nix-build-python3.9-aocd-1.1.1.drv-0/source
plugins: mock-3.6.1, freezegun-0.4.2, socket-0.4.0, requests-mock-1.9.3
[1mcollecting ... [0m[1m
collected 112 items                                                            [0m

tests/test_aocd.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31mF[0m[31m                                              [  7%][0m
tests/test_auth.py [31mF[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31m                                                 [ 11%][0m
tests/test_cookies.py [31mF[0m[31m                                                  [ 12%][0m
tests/test_date_finding.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31mF[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31m                                  [ 23%][0m
tests/test_date_introspection.py [31mF[0m[32m.[0m[31mF[0m[31mF[0m[31mF[0m[31m                                   [ 27%][0m
tests/test_get_data.py [32m.[0m[32m.[0m[32m.[0m[31mF[0m[31mF[0m[31mF[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31m                                        [ 36%][0m
tests/test_main.py [31mF[0m[32m.[0m[32m.[0m[31m                                                   [ 39%][0m
tests/test_models.py [32m.[0m[31mF[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31mF[0m[32m.[0m[32m.[0m[31mF[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31m                       [ 65%][0m
tests/test_runner.py [31mF[0m[31mF[0m[31mF[0m[31mF[0m[31mF[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31mF[0m[31mF[0m[31mF[0m[31mF[0m[31m                                      [ 77%][0m
tests/test_submit.py [32m.[0m[32m.[0m[31mF[0m[31mF[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31mF[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31mF[0m[32m.[0m[31m                                 [ 94%][0m
tests/test_utils.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31m                                               [100%][0m

=================================== FAILURES ===================================
[31m[1m______________________ test_attribute_errors_have_context ______________________[0m

    def test_attribute_errors_have_context():
>       with pytest.raises(AttributeError("nope")):
[1m[31mE       TypeError: 'AttributeError' object is not iterable[0m

[1m[31mtests/test_aocd.py[0m:69: TypeError
[31m[1m______________________________ test_no_session_id ______________________________[0m

test_token = PosixPath('/private/tmp/nix-build-python3.9-aocd-1.1.1.drv-0/pytest-of-_nixbld1/pytest-0/test_no_session_id0/.config/aocd-config/token')
capsys = <_pytest.capture.CaptureFixture object at 0x1049fc3a0>

    def test_no_session_id(test_token, capsys):
        test_token.unlink()
>       with pytest.raises(AocdError("Missing session ID")):
[1m[31mE       TypeError: 'AocdError' object is not iterable[0m

[1m[31mtests/test_auth.py[0m:9: TypeError
[31m[1m______________________________ test_cookie_scrape ______________________________[0m

mocker = <pytest_mock.plugin.MockerFixture object at 0x104a4d1f0>
capsys = <_pytest.capture.CaptureFixture object at 0x104a43df0>

    def test_cookie_scrape(mocker, capsys):
        mocker.patch("sys.argv", ["aocd-token", "--check"])
>       with pytest.raises(SystemExit(0)):
[1m[31mE       TypeError: 'SystemExit' object is not iterable[0m

[1m[31mtests/test_cookies.py[0m:8: TypeError
[31m[1m____________________________ test_year_out_of_range ____________________________[0m

freezer = <freezegun.api.FrozenDateTimeFactory object at 0x1048b2bb0>

    def test_year_out_of_range(freezer):
        freezer.move_to("2015-11-11")
>       with pytest.raises(AocdError("Time travel not supported yet")):
[1m[31mE       TypeError: 'AocdError' object is not iterable[0m

[1m[31mtests/test_date_finding.py[0m:28: TypeError
[31m[1m_______________ test_get_day_and_year_fail_no_filename_on_stack ________________[0m

    def test_get_day_and_year_fail_no_filename_on_stack():
>       with pytest.raises(AocdError("Failed introspection of filename")):
[1m[31mE       TypeError: 'AocdError' object is not iterable[0m

[1m[31mtests/test_date_introspection.py[0m:8: TypeError
[31m[1m____________________________ test_year_is_ambiguous ____________________________[0m

mocker = <pytest_mock.plugin.MockerFixture object at 0x1049bfb80>

    def test_year_is_ambiguous(mocker):
        fake_stack = [("~/2016/2017_q01.py",)]
        mocker.patch("aocd.get.traceback.extract_stack", return_value=fake_stack)
>       with pytest.raises(AocdError("Failed introspection of year")):
[1m[31mE       TypeError: 'AocdError' object is not iterable[0m

[1m[31mtests/test_date_introspection.py[0m:23: TypeError
[31m[1m_____________________________ test_day_is_unknown ______________________________[0m

mocker = <pytest_mock.plugin.MockerFixture object at 0x1048206d0>

    def test_day_is_unknown(mocker):
        fake_stack = [("~/2016.py",)]
        mocker.patch("aocd.get.traceback.extract_stack", return_value=fake_stack)
>       with pytest.raises(AocdError("Failed introspection of day")):
[1m[31mE       TypeError: 'AocdError' object is not iterable[0m

[1m[31mtests/test_date_introspection.py[0m:30: TypeError
[31m[1m_____________________________ test_day_is_invalid ______________________________[0m

mocker = <pytest_mock.plugin.MockerFixture object at 0x104a53f10>

    def test_day_is_invalid(mocker):
        fake_stack = [("~/2016/q27.py",)]
        mocker.patch("aocd.get.traceback.extract_stack", return_value=fake_stack)
>       with pytest.raises(AocdError("Failed introspection of day")):
[1m[31mE       TypeError: 'AocdError' object is not iterable[0m

[1m[31mtests/test_date_introspection.py[0m:37: TypeError
[31m[1m______________________________ test_server_error _______________________________[0m

requests_mock = <requests_mock.mocker.Mocker object at 0x104a0d0d0>
caplog = <_pytest.logging.LogCaptureFixture object at 0x104a016d0>

    def test_server_error(requests_mock, caplog):
        mock = requests_mock.get(
            url="https://adventofcode.com/2101/day/1/input",
            text="AWS meltdown",
            status_code=500,
        )
>       with pytest.raises(AocdError("Unexpected response")):
[1m[31mE       TypeError: 'AocdError' object is not iterable[0m

[1m[31mtests/test_get_data.py[0m:52: TypeError
[31m[1m________________________ test_puzzle_not_available_yet _________________________[0m

requests_mock = <requests_mock.mocker.Mocker object at 0x10482b5e0>
caplog = <_pytest.logging.LogCaptureFixture object at 0x1049bf970>

    def test_puzzle_not_available_yet(requests_mock, caplog):
        mock = requests_mock.get(
            url="https://adventofcode.com/2101/day/1/input",
            text="Not Found",
            status_code=404,
        )
>       with pytest.raises(PuzzleLockedError("2101/01 not available yet")):
[1m[31mE       TypeError: 'PuzzleLockedError' object is not iterable[0m

[1m[31mtests/test_get_data.py[0m:68: TypeError
[31m[1m_____________________ test_puzzle_not_available_yet_block ______________________[0m

requests_mock = <requests_mock.mocker.Mocker object at 0x1049d1ee0>
caplog = <_pytest.logging.LogCaptureFixture object at 0x1048115e0>
mocker = <pytest_mock.plugin.MockerFixture object at 0x1048cbe20>

    def test_puzzle_not_available_yet_block(requests_mock, caplog, mocker):
        mock = requests_mock.get(
            url="https://adventofcode.com/2101/day/1/input",
            text="Not Found",
            status_code=404,
        )
        blocker = mocker.patch("aocd._module.get.blocker")
>       with pytest.raises(PuzzleLockedError("2101/01 not available yet")):
[1m[31mE       TypeError: 'PuzzleLockedError' object is not iterable[0m

[1m[31mtests/test_get_data.py[0m:81: TypeError
[31m[1m____________________________ test_main_invalid_date ____________________________[0m

mocker = <pytest_mock.plugin.MockerFixture object at 0x1048d9880>
capsys = <_pytest.capture.CaptureFixture object at 0x104811550>

    def test_main_invalid_date(mocker, capsys):
        mocker.patch("sys.argv", ["aocd", "1", "2014"])
>       with pytest.raises(SystemExit(1)):
[1m[31mE       TypeError: 'SystemExit' object is not iterable[0m

[1m[31mtests/test_main.py[0m:8: TypeError
[31m[1m_________________________ test_get_answer_not_existing _________________________[0m

aocd_data_dir = PosixPath('/private/tmp/nix-build-python3.9-aocd-1.1.1.drv-0/pytest-of-_nixbld1/pytest-0/test_get_answer_not_existing0/.config/aocd-data')
requests_mock = <requests_mock.mocker.Mocker object at 0x104a629a0>

    def test_get_answer_not_existing(aocd_data_dir, requests_mock):
        requests_mock.get("https://adventofcode.com/2017/day/13")
        puzzle = Puzzle(day=13, year=2017)
>       with pytest.raises(AttributeError("answer_b")):
[1m[31mE       TypeError: 'AttributeError' object is not iterable[0m

[1m[31mtests/test_models.py[0m:26: TypeError
[31m[1m_____________________________ test_solve_no_plugs ______________________________[0m

mocker = <pytest_mock.plugin.MockerFixture object at 0x104acd7f0>

    def test_solve_no_plugs(mocker):
        mock = mocker.patch("pkg_resources.iter_entry_points", return_value=iter([]))
        puzzle = Puzzle(year=2018, day=1)
        expected = AocdError("Puzzle.solve is only available with unique entry point")
>       with pytest.raises(expected):
[1m[31mE       TypeError: 'AocdError' object is not iterable[0m

[1m[31mtests/test_models.py[0m:107: TypeError
[31m[1m_________________________ test_solve_for_unfound_user __________________________[0m

aocd_data_dir = PosixPath('/private/tmp/nix-build-python3.9-aocd-1.1.1.drv-0/pytest-of-_nixbld1/pytest-0/test_solve_for_unfound_user0/.config/aocd-data')
mocker = <pytest_mock.plugin.MockerFixture object at 0x1049d4070>

    def test_solve_for_unfound_user(aocd_data_dir, mocker):
        input_path = aocd_data_dir / "testauth.testuser.000" / "2018_01_input.txt"
        input_path.write_text("someinput")
        other_plug = mocker.Mock()
        other_plug.name = "otherplugin"
        mocker.patch("pkg_resources.iter_entry_points", return_value=iter([other_plug]))
        puzzle = Puzzle(year=2018, day=1)
>       with pytest.raises(AocdError("No entry point found for 'myplugin'")):
[1m[31mE       TypeError: 'AocdError' object is not iterable[0m

[1m[31mtests/test_models.py[0m:146: TypeError
[31m[1m____________________________ test_no_plugins_avail _____________________________[0m

capsys = <_pytest.capture.CaptureFixture object at 0x1047f5220>
mocker = <pytest_mock.plugin.MockerFixture object at 0x104a011c0>

    def test_no_plugins_avail(capsys, mocker):
        mock = mocker.patch("pkg_resources.iter_entry_points", return_value=iter([]))
        mocker.patch("sys.argv", ["aoc"])
        msg = (
            "There are no plugins available. Install some package(s) with a registered 'adventofcode.user' entry-point.\n"
            "See https://github.com/wimglenn/advent-of-code-sample for an example plugin package structure.\n"
        )
>       with pytest.raises(SystemExit(1)):
[1m[31mE       TypeError: 'SystemExit' object is not iterable[0m

[1m[31mtests/test_runner.py[0m:23: TypeError
[31m[1m____________________________ test_no_datasets_avail ____________________________[0m

capsys = <_pytest.capture.CaptureFixture object at 0x104a09fa0>
mocker = <pytest_mock.plugin.MockerFixture object at 0x104a563a0>
aocd_config_dir = PosixPath('/private/tmp/nix-build-python3.9-aocd-1.1.1.drv-0/pytest-of-_nixbld1/pytest-0/test_no_datasets_avail0/.config/aocd-config')

    def test_no_datasets_avail(capsys, mocker, aocd_config_dir):
        datasets_file = aocd_config_dir / "tokens.json"
        datasets_file.write_text("{}")
        mocker.patch("sys.argv", ["aoc"])
        msg = (
            "There are no datasets available to use.\n"
            "Either export your AOC_SESSION or put some auth tokens into {}\n".format(
                datasets_file
            )
        )
>       with pytest.raises(SystemExit(1)):
[1m[31mE       TypeError: 'SystemExit' object is not iterable[0m

[1m[31mtests/test_runner.py[0m:40: TypeError
[31m[1m__________________________________ test_main ___________________________________[0m

capsys = <_pytest.capture.CaptureFixture object at 0x1048cb130>
mocker = <pytest_mock.plugin.MockerFixture object at 0x104af0be0>
aocd_config_dir = PosixPath('/private/tmp/nix-build-python3.9-aocd-1.1.1.drv-0/pytest-of-_nixbld1/pytest-0/test_main0/.config/aocd-config')

    def test_main(capsys, mocker, aocd_config_dir):
        mock = mocker.patch("aocd.runner.run_for", return_value=0)
        ep1 = mocker.Mock()
        ep1.name = "user1"
        ep2 = mocker.Mock()
        ep2.name = "user2"
        mocker.patch("pkg_resources.iter_entry_points", return_value=iter([ep1, ep2]))
        datasets_file = aocd_config_dir / "tokens.json"
        datasets_file.write_text('{"data1": "token1", "data2": "token2"}')
        mocker.patch("sys.argv", ["aoc", "--years=2015", "--days", "3", "7"])
>       with pytest.raises(SystemExit(0)):
[1m[31mE       TypeError: 'SystemExit' object is not iterable[0m

[1m[31mtests/test_runner.py[0m:56: TypeError
[31m[1m_________________________________ test_results _________________________________[0m

mocker = <pytest_mock.plugin.MockerFixture object at 0x1048df610>
capsys = <_pytest.capture.CaptureFixture object at 0x104b1af10>

    def test_results(mocker, capsys):
        ep = mocker.Mock()
        ep.name = "testuser"
        ep.load.return_value = fake_entry_point
        mocker.patch("pkg_resources.iter_entry_points", return_value=iter([ep]))
        fake_puzzle = mocker.MagicMock()
        fake_puzzle.year = 2015
        fake_puzzle.day = 1
        fake_puzzle.input_data = "testinput"
        fake_puzzle.answer_a = "answer1"
        fake_puzzle.answer_b = "answer2"
        fake_puzzle.title = "The Puzzle Title"
        mocker.patch("aocd.runner.Puzzle", return_value=fake_puzzle)
        run_for(
            plugins=["testuser"],
            years=[2015],
            days=[1],
            datasets={"testdataset": "testtoken"},
        )
        ep.load.assert_called_once_with()
        out, err = capsys.readouterr()
        txt = "2015/1  - The Puzzle Title                           testuser/testdataset"
        assert txt in out
>       assert "part a: answer1 " in out
[1m[31mE       assert 'part a: answer1 ' in "\x1b[32m   0.97s\x1b[0m   2015/1  - The Puzzle Title                           testuser/testdataset   \x1b[31m✖\x1b[0m KeyError('_process_wrapper')\n"[0m

[1m[31mtests/test_runner.py[0m:111: AssertionError
[31m[1m______________________________ test_results_xmas _______________________________[0m

mocker = <pytest_mock.plugin.MockerFixture object at 0x105b7cf70>
capsys = <_pytest.capture.CaptureFixture object at 0x105b75ac0>

    def test_results_xmas(mocker, capsys):
        ep = mocker.Mock()
        ep.name = "testuser"
        ep.load.return_value = xmas_entry_point
        mocker.patch("pkg_resources.iter_entry_points", return_value=iter([ep]))
        fake_puzzle = mocker.MagicMock(
            year=2015,
            day=25,
            input_data="testinput",
            answer_a="answer1",
            answer_b="not_used",
            title="The Puzzle Title",
        )
        mocker.patch("aocd.runner.Puzzle", return_value=fake_puzzle)
        run_for(
            plugins=["testuser"],
            years=[2015],
            days=[25],
            datasets={"testdataset": "testtoken"},
        )
        ep.load.assert_called_once_with()
        out, err = capsys.readouterr()
        txt = "2015/25 - The Puzzle Title                           testuser/testdataset"
        assert txt in out
>       assert "part a: answer1 " in out
[1m[31mE       assert 'part a: answer1 ' in "\x1b[32m   0.77s\x1b[0m   2015/25 - The Puzzle Title                           testuser/testdataset   \x1b[31m✖\x1b[0m KeyError('_process_wrapper')\n"[0m

[1m[31mtests/test_runner.py[0m:140: AssertionError
[31m[1m________________________________ test_run_error ________________________________[0m

aocd_data_dir = PosixPath('/private/tmp/nix-build-python3.9-aocd-1.1.1.drv-0/pytest-of-_nixbld1/pytest-0/test_run_error0/.config/aocd-data')
mocker = <pytest_mock.plugin.MockerFixture object at 0x1051ae3a0>
capsys = <_pytest.capture.CaptureFixture object at 0x1051da280>

    def test_run_error(aocd_data_dir, mocker, capsys):
        title_path = aocd_data_dir / "titles"
        title_path.mkdir()
        title_file = title_path / "2018_25.txt"
        title_file.write_text("The Puzzle Title")
        input_path = aocd_data_dir / "testauth.testuser.000" / "2018_25_input.txt"
        input_path.write_text("someinput")
        answer_path = aocd_data_dir / "testauth.testuser.000" / "2018_25a_answer.txt"
        answer_path.write_text("answ")
        ep = mocker.Mock()
        ep.name = "testuser"
        ep.load.return_value = bugged_entry_point
        mocker.patch("pkg_resources.iter_entry_points", return_value=iter([ep]))
        run_for(
            plugins=["testuser"],
            years=[2018],
            days=[25],
            datasets={"default": "thetesttoken"},
        )
        out, err = capsys.readouterr()
        txt = "Exception(123, 456)"
        assert "✖" in out
>       assert txt in out
[1m[31mE       assert 'Exception(123, 456)' in "\x1b[32m   0.79s\x1b[0m   2018/25 - The Puzzle Title                           testuser/default   \x1b[31m✖\x1b[0m KeyError('_process_wrapper')\n"[0m

[1m[31mtests/test_runner.py[0m:200: AssertionError
[31m[1m___________________________ test_run_and_autosubmit ____________________________[0m

aocd_data_dir = PosixPath('/private/tmp/nix-build-python3.9-aocd-1.1.1.drv-0/pytest-of-_nixbld1/pytest-0/test_run_and_autosubmit0/.config/aocd-data')
mocker = <pytest_mock.plugin.MockerFixture object at 0x1051ae340>
capsys = <_pytest.capture.CaptureFixture object at 0x1058f4580>
requests_mock = <requests_mock.mocker.Mocker object at 0x1058f3250>

    def test_run_and_autosubmit(aocd_data_dir, mocker, capsys, requests_mock):
        title_path = aocd_data_dir / "titles"
        title_path.mkdir()
        title_file = title_path / "2015_01.txt"
        title_file.write_text("The Puzzle Title")
        input_path = aocd_data_dir / "testauth.testuser.000" / "2015_01_input.txt"
        input_path.write_text("testinput")
        answer_path = aocd_data_dir / "testauth.testuser.000" / "2015_01a_answer.txt"
        answer_path.write_text("answer1")
        requests_mock.get(url="https://adventofcode.com/2015/day/1")
        requests_mock.post(
            url="https://adventofcode.com/2015/day/1/answer",
            text="<article>That's not the right answer</article>",
        )
        ep = mocker.Mock()
        ep.name = "testuser"
        ep.load.return_value = fake_entry_point
        mocker.patch("pkg_resources.iter_entry_points", return_value=iter([ep]))
        run_for(
            plugins=["testuser"],
            years=[2015],
            days=[1],
            datasets={"default": "thetesttoken"},
        )
        out, err = capsys.readouterr()
>       assert "part a: answer1 " in out
[1m[31mE       assert 'part a: answer1 ' in "\x1b[32m   0.79s\x1b[0m   2015/1  - The Puzzle Title                           testuser/default   \x1b[31m✖\x1b[0m KeyError('_process_wrapper')\n"[0m

[1m[31mtests/test_runner.py[0m:229: AssertionError
[31m[1m__________________________ test_run_and_no_autosubmit __________________________[0m

aocd_data_dir = PosixPath('/private/tmp/nix-build-python3.9-aocd-1.1.1.drv-0/pytest-of-_nixbld1/pytest-0/test_run_and_no_autosubmit0/.config/aocd-data')
mocker = <pytest_mock.plugin.MockerFixture object at 0x1051e5700>
capsys = <_pytest.capture.CaptureFixture object at 0x1053dc520>
requests_mock = <requests_mock.mocker.Mocker object at 0x1053de220>

    def test_run_and_no_autosubmit(aocd_data_dir, mocker, capsys, requests_mock):
        title_path = aocd_data_dir / "titles"
        title_path.mkdir()
        title_file = title_path / "2015_01.txt"
        title_file.write_text("The Puzzle Title")
        input_path = aocd_data_dir / "testauth.testuser.000" / "2015_01_input.txt"
        input_path.write_text("testinput")
        answer_path = aocd_data_dir / "testauth.testuser.000" / "2015_01a_answer.txt"
        answer_path.write_text("answer1")
        requests_mock.get(url="https://adventofcode.com/2015/day/1")
        ep = mocker.Mock()
        ep.name = "testuser"
        ep.load.return_value = fake_entry_point
        mocker.patch("pkg_resources.iter_entry_points", return_value=iter([ep]))
        run_for(
            plugins=["testuser"],
            years=[2015],
            days=[1],
            datasets={"default": "thetesttoken"},
            autosubmit=False,
        )
        out, err = capsys.readouterr()
>       assert "part a: answer1 " in out
[1m[31mE       assert 'part a: answer1 ' in "\x1b[32m   0.82s\x1b[0m   2015/1  - The Puzzle Title                           testuser/default   \x1b[31m✖\x1b[0m KeyError('_process_wrapper')\n"[0m

[1m[31mtests/test_runner.py[0m:255: AssertionError
[31m[1m__________________________ test_load_input_from_file ___________________________[0m

mocker = <pytest_mock.plugin.MockerFixture object at 0x1053de370>

    def test_load_input_from_file(mocker):
        ep = mocker.Mock()
        ep.name = "file_ep_user"
        ep.load.return_value = file_entry_point
        a, b, walltime, error = run_one(2015, 1, "abcxyz", ep)
>       assert a == "123"
[1m[31mE       AssertionError: assert '' == '123'[0m
[1m[31mE         - 123[0m

[1m[31mtests/test_runner.py[0m:273: AssertionError
[31m[1m____________________________ test_submit_bogus_part ____________________________[0m

    def test_submit_bogus_part():
>       with pytest.raises(AocdError('part must be "a" or "b"')):
[1m[31mE       TypeError: 'AocdError' object is not iterable[0m

[1m[31mtests/test_submit.py[0m:37: TypeError
[31m[1m______________________________ test_server_error _______________________________[0m

requests_mock = <requests_mock.mocker.Mocker object at 0x105d61a00>
freezer = <freezegun.api.FrozenDateTimeFactory object at 0x105d60730>

    def test_server_error(requests_mock, freezer):
        freezer.move_to("2018-12-01 12:00:00Z")
        requests_mock.post(
            url="https://adventofcode.com/2018/day/1/answer", status_code=500
        )
>       with pytest.raises(AocdError("Non-200 response for POST: <Response [500]>")):
[1m[31mE       TypeError: 'AocdError' object is not iterable[0m

[1m[31mtests/test_submit.py[0m:46: TypeError
[31m[1m_____________ test_submit_correct_part_a_answer_for_part_b_blocked _____________[0m

requests_mock = <requests_mock.mocker.Mocker object at 0x105dba8b0>

    def test_submit_correct_part_a_answer_for_part_b_blocked(requests_mock):
        requests_mock.get(
            url="https://adventofcode.com/2018/day/1",
            text="<h2>Day 1: Yo Dawg</h2> <p>Your puzzle answer was <code>1234</code></p>",
        )
        requests_mock.post(
            url="https://adventofcode.com/2018/day/1/answer",
            text="<article>That's the right answer</article>",
        )
>       with pytest.raises(AocdError("cowardly refusing to re-submit answer_a (1234) for part b")):
[1m[31mE       TypeError: 'AocdError' object is not iterable[0m

[1m[31mtests/test_submit.py[0m:130: TypeError
[31m[1m__________________________ test_will_not_submit_null ___________________________[0m

    def test_will_not_submit_null():
>       with pytest.raises(AocdError("cowardly refusing to submit non-answer: None")):
[1m[31mE       TypeError: 'AocdError' object is not iterable[0m

[1m[31mtests/test_submit.py[0m:251: TypeError
[33m=============================== warnings summary ===============================[0m
tests/test_models.py:302
  /private/tmp/nix-build-python3.9-aocd-1.1.1.drv-0/source/tests/test_models.py:302: PytestUnknownMarkWarning: Unknown pytest.mark.answer_not_cached - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.answer_not_cached(install=False)

tests/test_models.py:310
  /private/tmp/nix-build-python3.9-aocd-1.1.1.drv-0/source/tests/test_models.py:310: PytestUnknownMarkWarning: Unknown pytest.mark.answer_not_cached - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.answer_not_cached(install=False)

tests/test_models.py:318
  /private/tmp/nix-build-python3.9-aocd-1.1.1.drv-0/source/tests/test_models.py:318: PytestUnknownMarkWarning: Unknown pytest.mark.answer_not_cached - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.answer_not_cached(install=False)

tests/test_models.py:326
  /private/tmp/nix-build-python3.9-aocd-1.1.1.drv-0/source/tests/test_models.py:326: PytestUnknownMarkWarning: Unknown pytest.mark.answer_not_cached - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.answer_not_cached(install=False)

tests/test_submit.py:255
  /private/tmp/nix-build-python3.9-aocd-1.1.1.drv-0/source/tests/test_submit.py:255: PytestUnknownMarkWarning: Unknown pytest.mark.answer_not_cached - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.answer_not_cached(rv='value')

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED tests/test_aocd.py::test_attribute_errors_have_context - TypeError: 'A...
FAILED tests/test_auth.py::test_no_session_id - TypeError: 'AocdError' object...
FAILED tests/test_cookies.py::test_cookie_scrape - TypeError: 'SystemExit' ob...
FAILED tests/test_date_finding.py::test_year_out_of_range - TypeError: 'AocdE...
FAILED tests/test_date_introspection.py::test_get_day_and_year_fail_no_filename_on_stack
FAILED tests/test_date_introspection.py::test_year_is_ambiguous - TypeError: ...
FAILED tests/test_date_introspection.py::test_day_is_unknown - TypeError: 'Ao...
FAILED tests/test_date_introspection.py::test_day_is_invalid - TypeError: 'Ao...
FAILED tests/test_get_data.py::test_server_error - TypeError: 'AocdError' obj...
FAILED tests/test_get_data.py::test_puzzle_not_available_yet - TypeError: 'Pu...
FAILED tests/test_get_data.py::test_puzzle_not_available_yet_block - TypeErro...
FAILED tests/test_main.py::test_main_invalid_date - TypeError: 'SystemExit' o...
FAILED tests/test_models.py::test_get_answer_not_existing - TypeError: 'Attri...
FAILED tests/test_models.py::test_solve_no_plugs - TypeError: 'AocdError' obj...
FAILED tests/test_models.py::test_solve_for_unfound_user - TypeError: 'AocdEr...
FAILED tests/test_runner.py::test_no_plugins_avail - TypeError: 'SystemExit' ...
FAILED tests/test_runner.py::test_no_datasets_avail - TypeError: 'SystemExit'...
FAILED tests/test_runner.py::test_main - TypeError: 'SystemExit' object is no...
FAILED tests/test_runner.py::test_results - assert 'part a: answer1 ' in "\x1...
FAILED tests/test_runner.py::test_results_xmas - assert 'part a: answer1 ' in...
FAILED tests/test_runner.py::test_run_error - assert 'Exception(123, 456)' in...
FAILED tests/test_runner.py::test_run_and_autosubmit - assert 'part a: answer...
FAILED tests/test_runner.py::test_run_and_no_autosubmit - assert 'part a: ans...
FAILED tests/test_runner.py::test_load_input_from_file - AssertionError: asse...
FAILED tests/test_submit.py::test_submit_bogus_part - TypeError: 'AocdError' ...
FAILED tests/test_submit.py::test_server_error - TypeError: 'AocdError' objec...
FAILED tests/test_submit.py::test_submit_correct_part_a_answer_for_part_b_blocked
FAILED tests/test_submit.py::test_will_not_submit_null - TypeError: 'AocdErro...
[31m================== [31m[1m28 failed[0m, [32m84 passed[0m, [33m5 warnings[0m[31m in 7.78s[0m[31m ===================[0m
```
</details>

--- a/maintainers/maintainer-list.nix
+++ b/maintainers/maintainer-list.nix
@@ -119,6 +119,12 @@
     githubId = 241628;
     name = "Adam Russell";
   };
+  aadibajpai = {
+    email = "hello@aadibajpai.com";
+    github = "aadibajpai";
+    githubId = 27063113;
+    name = "Aadi Bajpai";
+  };
   aanderse = {
     email = "aaron@fosslib.net";
     matrix = "@aanderse:nixos.dev";

--- a/pkgs/development/python-modules/pebble/default.nix
+++ b/pkgs/development/python-modules/pebble/default.nix
@@ -1,4 +1,4 @@
-{ lib, buildPythonPackage, isPy27, fetchPypi, pytestCheckHook }:
+{ lib, stdenv, buildPythonPackage, isPy27, fetchPypi, pytestCheckHook }:
 
 buildPythonPackage rec {
   pname = "pebble";
@@ -10,6 +10,8 @@ buildPythonPackage rec {
     inherit version;
     sha256 = "0a595f7mrf89xlck9b2x83bqybc9zd9jxkl0sa5cf19vax18rg8h";
   };
+
+  doCheck = !stdenv.isDarwin;
 
   checkInputs = [
     pytestCheckHook

--- a/pkgs/development/python-modules/pytest-raisin/default.nix
+++ b/pkgs/development/python-modules/pytest-raisin/default.nix
@@ -1,0 +1,37 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, flit-core
+, pytest
+}:
+
+buildPythonPackage rec {
+  pname = "pytest-raisin";
+  version = "0.3";
+  format = "flit";
+
+  src = fetchFromGitHub {
+    owner = "wimglenn";
+    repo = "pytest-raisin";
+    rev = "v${version}";
+    sha256 = "73cOrsqlE04m6X3a6VwtRzfi24oqkdO3HjKQH61bU88=";
+  };
+
+  nativeBuildInputs = [
+    flit-core
+  ];
+
+  propagatedBuildInputs = [
+    pytest
+  ];
+
+  # tests cause circular pytest-raisin already registered with pytest error
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Plugin enabling the use of exception instances with pytest.raises context";
+    homepage = "https://github.com/wimglenn/pytest-raisin";
+    license = licenses.mit;
+    maintainers = with maintainers; [ aadibajpai ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -7605,6 +7605,8 @@ in {
 
   pytest-raisesregexp = callPackage ../development/python-modules/pytest-raisesregexp { };
 
+  pytest-raisin = callPackage ../development/python-modules/pytest-raisin { };
+
   pytest-randomly = callPackage ../development/python-modules/pytest-randomly { };
 
   pytest-random-order = callPackage ../development/python-modules/pytest-random-order { };

